test/integration/targets/sqs_queue/tasks/main.yml

```diff
@@ -0,0 +1,114 @@
+---
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+- block:
+  - name: Test creating SQS queue
+    sqs_queue:
+      name: "{{ resource_prefix }}{{ 1000 | random }}"
+      <<: *aws_connection_info
+    register: create_result
+  - name: Assert SQS queue created
+    assert:
+      that:
+        - create_result.changed
+        - create_result.region == "{{ aws_region }}"
+  - name: Test deleting SQS queue
```

Comment:
Deletions should be kept in an `always` block so errors still allow resources to be cleaned up. Also, no need to `| random` since resource_prefix is always going to be unique to the test run.

Reply:
The random bit is because sqs won't let you re-create a queue with the same name for 60 seconds or so after it was deleted. `resource_prefix` is only unique per run, so since I'm spinning up and tearing down a few queues I need to make sure they don't use the same name.